### DISCUSSION TITLE
Specifying a default value for a member of a struct

### DISCUSSION THREAD

**Discussion by mitchgrout** [slint-ui/slint#4738] [Q&A]
As I understand, if a member of a struct is not explicitly initialized, it will be set to its default value. However, as far as I can tell there is no way to set a new default value for a member, as the following is illegal:

```js
export struct Test {
  value: int = 42
}
```

My take is that the default value for some types, including user-defined types, might not always be desirable. For example, I might want to have `export enum Accidental { flat, neutral, sharp }`. However, if I used this within a struct and forgot to explicitly initialize the field, then it would default to `flat` as it is the first element in the enum. My preference however would be that it should be `neutral`. 

Of course, I could resolve this by re-ordering my enum, but it feels as though there should be some flexibility here?

**Comment by tronical** [comment]
That sounds like a good idea to me as a suggestion.
